models/marts/core/int_all_graph_resources.sql

```diff
@@ -29,74 +29,82 @@ unioned_with_prefix as (
+        {{ dbt_utils.position('naming_convention_folders.folder_name_value','unioned_with_calc.directory_path') }} as position_folder,  
         nullif(column_name, '') as column_name,
         resource_name like 'unique%' and resource_type = 'test' as is_not_null_test,
         resource_name like 'not_null%' and resource_type = 'test' as is_unique_test,
-        unioned_with_prefix.is_enabled, 
-        unioned_with_prefix.materialized, 
-        unioned_with_prefix.on_schema_change, 
-        unioned_with_prefix.database, 
-        unioned_with_prefix.schema, 
-        unioned_with_prefix.package_name, 
-        unioned_with_prefix.alias, 
-        unioned_with_prefix.is_described, 
-        unioned_with_prefix.exposure_type, 
-        unioned_with_prefix.maturity, 
-        unioned_with_prefix.url, 
-        unioned_with_prefix.metric_type, 
-        unioned_with_prefix.model, 
-        unioned_with_prefix.label, 
-        unioned_with_prefix.sql, 
-        unioned_with_prefix.timestamp as timestamp,  
-        unioned_with_prefix.source_name, -- NULL for non-source resources
-        unioned_with_prefix.is_source_described, 
-        unioned_with_prefix.loaded_at_field, 
-        unioned_with_prefix.loader, 
-        unioned_with_prefix.identifier
+        unioned_with_calc.is_enabled, 
+        unioned_with_calc.materialized, 
+        unioned_with_calc.on_schema_change, 
+        unioned_with_calc.database, 
+        unioned_with_calc.schema, 
+        unioned_with_calc.package_name, 
+        unioned_with_calc.alias, 
+        unioned_with_calc.is_described, 
+        unioned_with_calc.exposure_type, 
+        unioned_with_calc.maturity, 
+        unioned_with_calc.url, 
+        unioned_with_calc.metric_type, 
+        unioned_with_calc.model, 
+        unioned_with_calc.label, 
+        unioned_with_calc.sql, 
+        unioned_with_calc.timestamp as timestamp,  
+        unioned_with_calc.source_name, -- NULL for non-source resources
+        unioned_with_calc.is_source_described, 
+        unioned_with_calc.loaded_at_field, 
+        unioned_with_calc.loader, 
+        unioned_with_calc.identifier
 
-    from unioned_with_prefix
+    from unioned_with_calc
     left join naming_convention_prefixes
-        on unioned_with_prefix.prefix = naming_convention_prefixes.prefix_value
+        on unioned_with_calc.prefix = naming_convention_prefixes.prefix_value
 
-    left join naming_convention_folders
-        -- we added '/' at the front of the path, to be /folder1/folder2/folder3/mmodel_name.sql
-        -- and we search for /naming_convention_folder/ ; this allows us to only match full folder names
-        on file_path_prefixed_with_slash like {{ dbt_utils.concat(["'%/'",'naming_convention_folders.folder_name_value',"'/%'"]) }}
-    
-    where coalesce(unioned_with_prefix.is_enabled, True) = True
-    and unioned_with_prefix.package_name != 'dbt_project_evaluator'
+    cross join naming_convention_folders   
```

Comment:
why is this cross join necessary?

Reply:
So that we can check the position of all `model_types` in the `file_path`...
It's the example where you have a marts model `my_model` nested BOTH in the marts AND intermediate folder.
`file_path` = `models/marts/intermediate/my_model.sql`
This cross join allows us to handle those situations and correctly identify that this should be an intermediate model (not a marts model).